src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java

```diff
@@ -0,0 +1,34 @@
+package com.example.simplefileshare.simplefileshare.controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.example.simplefileshare.simplefileshare.services.AWSService;
+import com.example.simplefileshare.simplefileshare.error.utils.ErrorUtils;
+import com.example.simplefileshare.simplefileshare.data.models.base.BaseResponse;
+import com.amazonaws.SdkClientException;
+import java.io.FileNotFoundException;
+
+
+@RestController
+@RequestMapping("/api/files")
+public class FileController {
+
+    @Autowired
+    private AWSService uploadService;
+
+    @PostMapping
+    public String upload(@RequestBody MultipartFile file) throws Exception{
+        try{
+            if(!file) throw new FileNotFoundException("Invalid File");
+            String url = uploadService.uploadMultipartFile(file);
+            return new ResponseEntity<>(new BaseResponse<>(null, url), HttpStatus.CREATED);
+        } catch (SdkClientException e){
+            throw ErrorUtils.createApiError(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase(), "Some error occurred");
+        } catch (FileNotFoundException e){
+            throw ErrorUtils.createApiError(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase(), "Some file error occurred");
```

Comment:
We should throw a Bad request error in this case.

Reply:
Added the changes.